### PR TITLE
[Feature] Moving to GitHub Actions

### PR DESCRIPTION
**Pull request by GuyAv46**
**Describe the changes in the pull request**

**Which issues this PR fixes**
1. #3794 
2. #3904 
3. #3984
4. #3908 
5. #4012 
6. #4024 
7. #4000 
8. #4045 
9. #4096 

**Main objects this PR modified**
1. CI

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

**Comment by alonre24** [comment]
Since many `.yml` were created, consider creating some sub-directories (for flows, tasks, benchmarks for example)

**Comment by GuyAv46** [comment]
@alonre24 I wish it was possible. Workflows can only be placed directly in `.github/workflows`

**Comment by alonre24** [comment]
> @alonre24 I wish it was possible. Workflows can only be placed directly in `.github/workflows`

What about the sub-tasks that are always used by `workflow call`?

**Comment by GuyAv46** [comment]
> > @alonre24 I wish it was possible. Workflows can only be placed directly in `.github/workflows`
> 
> What about the sub-tasks that are always used by `workflow call`?

they are still workflow. That is why I added the naming prefix - so they at least grouped together.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1071,28 +1071,28 @@ workflows:
       << pipeline.parameters.run_default_flow >>
     jobs:
       - build-linux-debian:
-          <<: *not-on-integ-branch
```
Comment (alonre24) [inline .circleci/config.yml:1074]: The nightly flows will still be running in the meantime - on purpose?

File: src/version.h [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
 // This is where the modules build/version is declared.
 // If declared with -D in compile time, this file is ignored
 
 #pragma once
 
-#define REDISEARCH_VERSION_MAJOR 99
-#define REDISEARCH_VERSION_MINOR 99
-#define REDISEARCH_VERSION_PATCH 99
+#define REDISEARCH_VERSION_MAJOR 46
```
Comment (alonre24) [inline src/version.h:13]: remember to change back before merge

**Comment by filipecosta90** [comment]
### Automated performance analysis summary

This comment was automatically generated given there is performance data available.

In summary:
- Detected a total of 28 stable tests between versions.
- Detected a total of 4 highly unstable benchmarks.
- Detected a total of 2 improvements above the improvement water line.
- Detected a total of 9 regressions bellow the regression water line 5.0.

You can check a comparison in detail via the [grafana link](https://benchmarksrediscom.grafana.net/d//3Ejv2wZnk/?var-branch=master&var-branch=feature-github-actions)

### Comparison between master and feature-github-actions.

Time Period from 30 days ago. (environment used: oss-standalone)

|                                           Test Case                                           |Baseline master (median obs. +- std.dev)|Comparison feature-github-actions (median obs. +- std.dev)|% change (higher-better)|                Note                 |
|-----------------------------------------------------------------------------------------------|----------------------------------------|---------------------------------------------------------:|------------------------|-------------------------------------|
|ftsb-10K-enwiki_abstract-hashes-fulltext-sortby                                                | 69 +- 1.4% (7 datapoints)              |                                                        68|-1.2%                   |No Change                            |
|ftsb-10K-enwiki_abstract-hashes-term-prefix                                                    | 7351 +- 1.7% (7 datapoints)            |                                                      7249|-1.4%                   |No Change                            |
|ftsb-10K-enwiki_abstract-hashes-term-suffix                                                    | 2072 +- 2.0% (7 datapoints)            |                                                      2162|4.3%                    |potential IMPROVEMENT                |
|ftsb-10K-enwiki_abstract-hashes-term-suffix-withsuffixtrie                                     | 68355 +- 5.1% (7 datapoints)           |                                                     64533|-5.6%                   |waterline=5.1%. REGRESSION           |
|ftsb-10K-enwiki_abstract-hashes-term-wildcard                                                  | 12308 +- 1.9% (7 datapoints)           |                                                     12436|1.0%                    |No Change                            |
|ftsb-10K-enwiki_pages-hashes-fulltext-mixed_simple-1word-query_write_1_to_read_20.yml          | 1253 +- 1.7% (7 datapoints)            |                                                      1111|-11.3%                  |REGRESSION                           |
|ftsb-10K-enwiki_pages-hashes-load                                                              | 51938 +- 4.4% (7 datapoints)           |                                                     47853|-7.9%                   |REGRESSION                           |
|ftsb-1K-enwiki_abstract-hashes-term-contains                                                   | 1763 +- 2.0% (7 datapoints)            |                                                      1842|4.5%                    |potential IMPROVEMENT                |
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-intersection-query                               | 359 +- 8.1% (7 datapoints)             |                                                       383|6.8%                    |waterline=8.1%. potential IMPROVEMENT|
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-intersection-query-non-sortable                  | 42 +- 13.9% UNSTABLE (7 datapoints)    |                                                        32|-23.7%                  |UNSTABLE (very high variance)        |
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-intersection-query@100_ops_sec                   | 100 +- 0.0% (7 datapoints)             |                                                       100|0.0%                    |No Change                            |
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-union-query                                      | 2671 +- 3.6% (7 datapoints)            |                                                      2710|1.4%                    |No Change                            |
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-union-query-non-sortable                         | 1056 +- 5.3% (7 datapoints)            |                                                       951|-9.9%                   |waterline=5.3%. REGRESSION           |
|ftsb-1M-enwiki_abstract-hashes-fulltext-2word-union-query@100_ops_sec                          | 100 +- 0.0% (7 datapoints)             |                                                       100|-0.0%                   |No Change                            |
|ftsb-1M-enwiki_abstract-hashes-fulltext-simple-1word-query                                     | 1033 +- 8.2% (7 datapoints)            |                                                       797|-22.9%                  |waterline=8.2%. REGRESSION           |
|ftsb-1M-enwiki_abstract-hashes-fulltext-simple-1word-query@100_ops_sec                         | 100 +- 0.0% (7 datapoints)             |                                                       100|0.0%                    |No Change                            |
|ftsb-1M-enwiki_abstract-hashes-load                                                            | 23110 +- 1.8% (7 datapoints)           |                                                     23178|0.3%                    |No Change                            |
|ftsb-1M-nyc_taxis-ftadd-load                                                                   | 22206 +- 2.2% (7 datapoints)           |                                                     22252|0.2%                    |No Change                            |
|ftsb-1M-nyc_taxis-hashes-load                                                                  | 22643 +- 2.3% (7 datapoints)           |                                                     22867|1.0%                    |No Change                            |
|search-aggregate-post-filter-simple.yml                                                        | 93801 +- 7.0% (7 datapoints)           |                                                     86154|-8.2%                   |waterline=7.0%. REGRESSION           |
|search-filtering-tag-numeric                                                                   | 188 +- 8.3% (7 datapoints)             |                                                       159|-15.6%                  |waterline=8.3%. REGRESSION           |
|search-filtering-tag-numeric-filter-pipeline                                                   | 19074 +- 2.0% (7 datapoints)           |                                                     19469|2.1%                    |No Change                            |
|search-ftsb-10K-enwiki_abstract-hashes-term-withoutsuffix-trie                                 | 41007 +- 4.5% (7 datapoints)           |                                                     37579|-8.4%                   |REGRESSION                           |
|search-ftsb-10K-enwiki_abstract-hashes-term-withsuffix-trie                                    | 40987 +- 2.9% (7 datapoints)           |                                                     37668|-8.1%                   |REGRESSION                           |
|search-ftsb-1700K-docs-union-iterators-q3                                                      | 6 +- 1.8% (7 datapoints)               |                                                         6|-3.1%                   |potential REGRESSION                 |
|search-ftsb-1M-enwiki_abstract-hashes-fulltext-2word-intersection-query-non-sortable@50_ops_sec| 50 +- 6.0% (7 datapoints)              |                                                        50|-0.4%                   |waterline=6.0%. No Change            |
|search-ftsb-1M-enwiki_abstract-hashes-fulltext-2word-union-query-non-sortable@100_ops_sec      | 100 +- 0.0% (7 datapoints)             |                                                       100|-0.0%                   |No Change                            |
|search-ftsb-1M-enwiki_abstract-hashes-fulltext-simple-1word-query-non-sortable                 | 150 +- 3.7% (7 datapoints)             |                                                       156|3.4%                    |potential IMPROVEMENT                |
|search-ftsb-370K-docs-union-iterators-q4                                                       | 6 +- 0.9% (7 datapoints)               |                                                         6|0.5%                    |No Change                            |
|search-ftsb-5200K-docs-union-iterators-q1                                                      | 1 +- 0.9% (7 datapoints)               |                                                         1|-1.9%                   |No Change                            |
|search-ftsb-5500K-docs-union-iterators-q2                                                      | 0 +- 2.2% (7 datapoints)               |                                                         0|0.0%                    |                                     |
|search-geo                                                                                     | 185 +- 3.6% (7 datapoints)             |                                                       183|-1.0%                   |No Change                            |
|search-high-cardinality-negation-term-baseline                                                 | 20 +- 3.1% (7 datapoints)              |                                                        20|-0.7%                   |No Change                            |
|search-high-cardinality-negation-term-comparison_union_all_other_terms                         | 7 +- 2.4% (7 datapoints)               |                                                         7|-3.5%                   |potential REGRESSION                 |
|search-numeric                                                                                 | 3104 +- 29.5% UNSTABLE (7 datapoints)  |                                                      3273|5.4%                    |UNSTABLE (very high variance)        |
|search-numeric-optimize                                                                        | 8977 +- 1.5% (7 datapoints)            |                                                      8991|0.2%                    |No Change                            |
|search-numeric-sortby                                                                          | 3229 +- 24.7% UNSTABLE (7 datapoints)  |                                                      2780|-13.9%                  |UNSTABLE (very high variance)        |
|search-numeric-sortby-desc                                                                     | 3151 +- 8.3% (7 datapoints)            |                                                      5207|65.3%                   |waterline=8.3%. IMPROVEMENT          |
|search-numeric-sortby-desc-optimize                                                            | 35 +- 21.1% UNSTABLE (7 datapoints)    |                                                        55|55.9%                   |UNSTABLE (very high variance)        |
|search-numeric-sortby-optimize                                                                 | 20 +- 3.5% (7 datapoints)              |                                                        21|2.6%                    |No Change                            |
|vecsim-arxiv-titles-384-angular-filters-m16-ef-128-fulltext-filter                             | 492 +- 2.4% (7 datapoints)             |                                                       488|-0.8%                   |No Change                            |
|vecsim-arxiv-titles-384-angular-filters-m16-ef-128-numeric-filter                              | 108 +- 3.4% (7 datapoints)             |                                                       107|-1.6%                   |No Change                            |
|vecsim-arxiv-titles-384-angular-filters-m16-ef-128-tag-filter                                  | 55584 +- 4.9% (7 datapoints)           |                                                     59223|6.5%                    |IMPROVEMENT                          |

**Review by alonre24 — APPROVED** [review]

**Pull request merged**